crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -7806,16 +7800,21 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
         let slice_node = subscript.slice.as_ref();
         let call_argument_types = match slice_node {
             ast::Expr::Tuple(tuple) => {
-                let arguments = CallArgumentTypes::positional(
+                let arguments = CallArguments::positional(
                     tuple.elts.iter().map(|elt| self.infer_type_expression(elt)),
                 );
                 self.store_expression_type(
                     slice_node,
-                    TupleType::from_elements(self.db(), arguments.iter().map(|(_, ty)| ty)),
+                    TupleType::from_elements(
+                        self.db(),
+                        arguments
+                            .iter()
+                            .map(|(_, ty)| ty.unwrap_or_else(Type::unknown)),
+                    ),
```

Comment:
Nit: You could add a `arguments.types()` method that returns an iterator over types